gateway/src/main/java/io/zeebe/gateway/impl/RequestManager.java

```diff
@@ -144,43 +171,49 @@ private void updateTopologyAndDeterminePartition(
         });
   }
 
-  private Supplier<Integer> determineRemoteProvider(RequestResponseHandler requestHandler) {
-    if (!requestHandler.addressesSpecificTopic() && !requestHandler.addressesSpecificPartition()) {
-      return new BrokerProvider(ClusterStateImpl::getRandomBroker);
-    } else {
-      final int targetPartition;
-      if (!requestHandler.addressesSpecificPartition()) {
-        int proposedPartition =
-            dispatchStrategy.determinePartition(requestHandler.getTargetTopic());
-
-        if (proposedPartition >= 0) {
-          targetPartition = proposedPartition;
-        } else {
-          final ActorFuture<Integer> partitionFuture =
-              updateTopologyAndDeterminePartition(requestHandler.getTargetTopic());
-
-          try {
-            proposedPartition =
-                partitionFuture.get(requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
-          } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            proposedPartition = -1;
-          }
-
-          targetPartition = proposedPartition;
-        }
-        if (targetPartition < 0) {
-          throw new ClientException(
-              "Cannot determine target partition for request. Request was: "
-                  + requestHandler.describeRequest());
-        } else {
-          requestHandler.onSelectedPartition(targetPartition);
-        }
+  private ActorFuture<Supplier<Integer>> determineRemoteProvider(
```

Comment:
Do we have tests which checks if the determineRemoteProvider works? Or that the requests are send to the correct partitions?